src/classes/RD2_RecurringDonation.cls

```diff
@@ -169,14 +169,20 @@ public inherited sharing class RD2_RecurringDonation {
     /**
     * @description Revises and modifies the Recurring Donation status
     * @param currentDate Used to determined current date
+    * @param scheduleService RD Schedule Service
     * @return RD2_RecurringDonation This record instance
     */
-    public RD2_RecurringDonation reviseStatus(Date currentDate) {
+    public RD2_RecurringDonation reviseStatus(Date currentDate, RD2_ScheduleService scheduleService) {
+        if (!isActive()) {
+            return this;
+        }
+
         if (isFixedLength()) {
             reviseStatusForFixedLength();
 
         } else if (isOpenEnded()) {
             reviseStatusForOpenEnded(currentDate);
+            reviseStatusForPaused(currentDate, scheduleService);
```

Comment:
I think even the fixed RD can be paused, so the check would apply to both fixed and open RDs?